Exception while starting

```
Exception in thread "HmIP Network Key Thread" java.lang.IllegalStateException: Unable to acquire the Scheduler
	at org.osgi.util.pushstream.PushStreamProvider.acquireScheduler(PushStreamProvider.java:72)
	at org.osgi.util.pushstream.PushStreamProvider.createSimplePushEventSource(PushStreamProvider.java:550)
	at org.osgi.util.pushstream.PushStreamProvider.createSimpleEventSource(PushStreamProvider.java:491)
	at in.bytehue.messaging.mqtt5.provider.MessageSubscriptionProvider.subscribe(MessageSubscriptionProvider.java:124)
	at in.bytehue.messaging.mqtt5.provider.MessageSubscriptionProvider.subscribe(MessageSubscriptionProvider.java:102)
	at in.bytehue.messaging.mqtt5.provider.MessageReplyToPublisherProvider.publishWithReply(MessageReplyToPublisherProvider.java:129)
	at in.bytehue.messaging.mqtt5.provider.MessageReplyToPublisherProvider.publishWithReply(MessageReplyToPublisherProvider.java:119)
	at com.qivicon.runtime.grappa.wot.impl.MqttRequestDispatcherImpl.doRequest(MqttRequestDispatcherImpl.java:63)
	at com.qivicon.runtime.grappa.wot.impl.CloudConfigurationServiceImpl.putConfiguration(CloudConfigurationServiceImpl.java:97)
	at com.qivicon.security.homematicip.impl.NetworkKeyManager.storeInBackend(NetworkKeyManager.java:304)
	at com.qivicon.security.homematicip.impl.NetworkKeyManager.checkKeyInBackend(NetworkKeyManager.java:177)
	at com.qivicon.security.homematicip.HomematicIpSecurityComponent$NetworkKeyCreator.run(HomematicIpSecurityComponent.java:244)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireInterruptibly(AbstractQueuedSynchronizer.java:1220)
	at java.util.concurrent.locks.ReentrantLock.lockInterruptibly(ReentrantLock.java:335)
	at org.osgi.util.pushstream.PushStreamProvider.acquireScheduler(PushStreamProvider.java:59)
	... 12 more
```

--- a/in.bytehue.messaging.mqtt5.provider/src/main/java/in/bytehue/messaging/mqtt5/provider/MessagePublisherProvider.java
+++ b/in.bytehue.messaging.mqtt5.provider/src/main/java/in/bytehue/messaging/mqtt5/provider/MessagePublisherProvider.java
@@ -109,7 +109,7 @@ public final class MessagePublisherProvider implements MessagePublisher {
 		publish(message, context, null);
 	}
 
-	private void publish(final Message message, MessageContext context, String channel) {
+	private synchronized void publish(final Message message, MessageContext context, String channel) {
 		try {
 			if (context == null) {
 				context = message.getContext();

--- a/in.bytehue.messaging.mqtt5.provider/src/main/java/in/bytehue/messaging/mqtt5/provider/MessageReplyToPublisherProvider.java
+++ b/in.bytehue.messaging.mqtt5.provider/src/main/java/in/bytehue/messaging/mqtt5/provider/MessageReplyToPublisherProvider.java
@@ -87,23 +87,23 @@ public final class MessageReplyToPublisherProvider implements ReplyToPublisher, 
     }
     //@formatter:on
 
-    @Reference(service = LoggerFactory.class)
-    private Logger logger;
+	@Reference(service = LoggerFactory.class)
+	private Logger logger;
 
-    @Reference
-    private MessagePublisherProvider publisher;
+	@Reference
+	private MessagePublisherProvider publisher;
 
-    @Reference
-    private MessageSubscriptionProvider subscriber;
+	@Reference
+	private MessageSubscriptionProvider subscriber;
 
-    @Activate
-    private BundleContext bundleContext;
+	@Activate
+	private BundleContext bundleContext;
 
-    private final PromiseFactory promiseFactory;
+	private final PromiseFactory promiseFactory;
 
-    @Activate
-    public MessageReplyToPublisherProvider(final Config config) {
-        //@formatter:off
+	@Activate
+	public MessageReplyToPublisherProvider(final Config config) {
+		//@formatter:off
         final ThreadFactory threadFactory =
                 new ThreadFactoryBuilder()
                         .setThreadFactoryName(config.threadNamePrefix())
@@ -112,63 +112,65 @@ public final class MessageReplyToPublisherProvider implements ReplyToPublisher, 
                         .build();
         promiseFactory = new PromiseFactory(newFixedThreadPool(config.numThreads(), threadFactory));
         //@formatter:on
-    }
+	}
 
-    @Override
-    public Promise<Message> publishWithReply(final Message requestMessage) {
-        return publishWithReply(requestMessage, requestMessage.getContext());
-    }
+	@Override
+	public Promise<Message> publishWithReply(final Message requestMessage) {
+		return publishWithReply(requestMessage, requestMessage.getContext());
+	}
 
-    @Override
-    public Promise<Message> publishWithReply(final Message requestMessage, final MessageContext replyToContext) {
-        final Deferred<Message> deferred = promiseFactory.deferred();
-        final ReplyToDTO dto = new ReplyToDTO(requestMessage, replyToContext);
+	@Override
+	public synchronized Promise<Message> publishWithReply(final Message requestMessage,
+			final MessageContext replyToContext) {
+		final Deferred<Message> deferred = promiseFactory.deferred();
+		final ReplyToDTO dto = new ReplyToDTO(requestMessage, replyToContext);
 
-        publisher.publish(requestMessage, dto.pubChannel);
-        // resolve the promise on first response
-        subscriber.subscribe(dto.subChannel).forEach(deferred::resolve);
-        return deferred.getPromise();
-    }
+		publisher.publish(requestMessage, dto.pubChannel);
+		// resolve the promise on first response
+		subscriber.subscribe(dto.subChannel).forEach(deferred::resolve);
+		return deferred.getPromise();
+	}
 
-    @Override
-    public PushStream<Message> publishWithReplyMany(final Message requestMessage) {
-        return publishWithReplyMany(requestMessage, requestMessage.getContext());
-    }
+	@Override
+	public PushStream<Message> publishWithReplyMany(final Message requestMessage) {
+		return publishWithReplyMany(requestMessage, requestMessage.getContext());
+	}
 
-    @Override
-    public PushStream<Message> publishWithReplyMany(final Message requestMessage, final MessageContext replyToContext) {
-        final ReplyToDTO dto = new ReplyToDTO(requestMessage, replyToContext);
+	@Override
+	public synchronized PushStream<Message> publishWithReplyMany(final Message requestMessage,
+			final MessageContext replyToContext) {
+		final ReplyToDTO dto = new ReplyToDTO(requestMessage, replyToContext);
 
-        publisher.publish(requestMessage, dto.pubChannel);
-        return subscriber.subscribe(dto.subChannel);
-    }
+		publisher.publish(requestMessage, dto.pubChannel);
+		return subscriber.subscribe(dto.subChannel);
+	}
 
-    private class ReplyToDTO {
-        String pubChannel;
-        String subChannel;
+	private class ReplyToDTO {
+		String pubChannel;
+		String subChannel;
 
-        ReplyToDTO(final Message message, final MessageContext context) {
-            autoGenerateCorrelationIdIfAbsent(message);
-            autoGenerateReplyToChannelIfAbsent(message);
+		ReplyToDTO(final Message message, final MessageContext context) {
+			autoGenerateCorrelationIdIfAbsent(message);
+			autoGenerateReplyToChannelIfAbsent(message);
 
-            pubChannel = context.getChannel();
-            subChannel = context.getReplyToChannel();
-        }
+			pubChannel = context.getChannel();
+			subChannel = context.getReplyToChannel();
+		}
 
-        private void autoGenerateCorrelationIdIfAbsent(final Message message) {
-            final MessageContextProvider context = (MessageContextProvider) message.getContext();
-            context.correlationId = getCorrelationId(context, bundleContext, logger);
-        }
+		private void autoGenerateCorrelationIdIfAbsent(final Message message) {
+			final MessageContextProvider context = (MessageContextProvider) message.getContext();
+			context.correlationId = getCorrelationId(context, bundleContext, logger);
+		}
 
-        private void autoGenerateReplyToChannelIfAbsent(final Message message) {
-            final MessageContextProvider context = (MessageContextProvider) message.getContext();
+		private void autoGenerateReplyToChannelIfAbsent(final Message message) {
+			final MessageContextProvider context = (MessageContextProvider) message.getContext();
 
-            if (context.getReplyToChannel() == null) {
-                context.replyToChannel = UUID.randomUUID().toString();
-                logger.info("Auto-generated reply-to channel '{}' as it is missing in the request",
-                        context.replyToChannel);
-            }
-        }
-    }
+			if (context.getReplyToChannel() == null) {
+				context.replyToChannel = UUID.randomUUID().toString();
+				logger.info("Auto-generated reply-to channel '{}' as it is missing in the request",
+						context.replyToChannel);
+			}
+		}
+	}
 
 }

--- a/in.bytehue.messaging.mqtt5.provider/src/main/java/in/bytehue/messaging/mqtt5/provider/MessageSubscriptionProvider.java
+++ b/in.bytehue.messaging.mqtt5.provider/src/main/java/in/bytehue/messaging/mqtt5/provider/MessageSubscriptionProvider.java
@@ -114,11 +114,11 @@ public final class MessageSubscriptionProvider implements MessageSubscription {
         return subscribe(null, subChannel, pubChannel, reference);
     }
 
-    private PushStream<Message> subscribe(
-            MessageContext context,
-            final String subChannel,
-            final String pubChannel,
-            final ServiceReference<?> reference) {
+    private synchronized PushStream<Message> subscribe(
+	            MessageContext context,
+	            final String subChannel,
+	            final String pubChannel,
+	            final ServiceReference<?> reference) {
 
         final PushStreamProvider provider = new PushStreamProvider();
         final SimplePushEventSource<Message> source = provider.createSimpleEventSource(Message.class);

